### PR TITLE
Timeline: Pop-out effect and sorting

### PR DESCRIPTION
**Pull request by aplett98**
Large diff because I had to modify the data again.

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -30,6 +30,7 @@ function createVis(error, matrixData, ageData, timeData) {
   // clean up time data
   timeData.forEach(d => {
     d.same_sex = +d.same_sex === -1 ? null : +d.same_sex === 1;
+    d.age = +d.age;
```
Comment (jebking) [inline js/main.js:33]: good catch, is there anything else that would need filtered out like that?

File: js/timelinevis.js [hunk 2]
```diff
@@ -18,6 +18,7 @@ TimelineVis = function(_parentElement, _data) {
 
 TimelineVis.prototype.initVis = function() {
     var vis = this;
+    vis.r = 4;
```
Comment (jebking) [inline js/timelinevis.js:21]: sort of a magic number but it looks like it means the radius

File: js/timelinevis.js [hunk 3]
```diff
@@ -35,15 +36,16 @@ TimelineVis.prototype.initVis = function() {
...
+        .attr("overflow", "auto");
 
     // SVG clipping path
-    vis.svg.append("defs")
-        .append("clipPath")
-        .attr("id", "clip")
-        .append("rect")
-        .attr("width", vis.width + 4)
-        .attr("height", vis.height + 4);
+    // vis.svg.append("defs")
```
Comment (jebking) [inline js/timelinevis.js:43]: i guess you're leaving that in for later?

File: js/timelinevis.js [hunk 4]
```diff
@@ -82,7 +69,9 @@ TimelineVis.prototype.initVis = function() {
 
 TimelineVis.prototype.wrangleData = function() {
     var vis = this;
-
+    vis.data = vis.data.sort((a, b) => a.dates[0].date - b.dates[0].date);
+    vis.data = vis.data.filter((_, i) => i < 100);
```
Comment (jebking) [inline js/timelinevis.js:75]: so it's only taking the first 100

File: js/timelinevis.js [hunk 5]
```diff
@@ -94,8 +83,12 @@ TimelineVis.prototype.wrangleData = function() {
 
 TimelineVis.prototype.updateVis = function() {
     var vis = this;
+    vis.y.domain([0, vis.data.length-1]);
+    console.log(vis.y.range());
+    vis.x.domain([d3.min(vis.data, d => d.dates[0].date), d3.max(vis.data, d => d.dates[d.dates.length-1].date)]);
```
Comment (jebking) [inline js/timelinevis.js]: hard to read

File: js/timelinevis.js [hunk 6]
```diff
@@ -94,8 +83,12 @@ TimelineVis.prototype.wrangleData = function() {
...
 
     vis.xAxis = d3.axisBottom()
+        .ticks(5)
```
Comment (jebking) [inline js/timelinevis.js:91]: magic number,  sort of.

File: js/timelinevis.js [hunk 7]
```diff
@@ -116,15 +108,15 @@ TimelineVis.prototype.updateVis = function() {
...
 
 
     timelineGroup.append("path")
-        .attr("stroke", "lightgrey")
-        .attr("stroke-width", 1)
+        .attr("stroke", "black")
+        .attr("stroke-width", 1.5)
         .merge(timeline)
-        .attr("d", d => `M${vis.x(d.dates[0].date)} 0 L${vis.x(d.dates[d.dates.length-1].date)} 0 Z`);
+        .attr("d", d => `M${vis.x(d.dates[0].date)+4} 0 L${vis.x(d.dates[d.dates.length-1].date)-4} 0 Z`);
```
Comment (jebking) [inline js/timelinevis.js:119]: might be nice to include a comment saying what this does (draw a line?)

File: js/timelinevis.js [hunk 8]
```diff
@@ -133,13 +125,46 @@ TimelineVis.prototype.updateVis = function() {
...
+            d3.select(this.parentNode)
+                .raise()
+                .selectAll("path")
+                .attr("opacity", 1);
+            d3.select(this.parentNode)
+                .selectAll(".date")
+                .attr("opacity", 1)
+                // .transition()
+                // .duration(750)
+                .attr("r", 7);
```
Comment (jebking) [inline js/timelinevis.js]: magic number, and isn't referring back to `vis.r` that was defined earlier

File: js/timelinevis.js [hunk 9]
```diff
@@ -133,13 +125,46 @@ TimelineVis.prototype.updateVis = function() {
...
         .transition()
         .attr("fill", d => vis.color(d.milestone))
         .attr("cx", d => vis.x(d.date));
     date.exit().remove();
 
 };
+
+
+function relationshipLength(d) {
+    return (d.dates[d.dates.length-1].date - d.dates[0].date);
```
Comment (jebking) [inline js/timelinevis.js:168]: nice, maybe elucidate how the data is structured

File: js/timelinevis.js [hunk 10]
```diff
@@ -133,13 +125,46 @@ TimelineVis.prototype.updateVis = function() {
     date.enter().append("circle")
         .attr("class", "date")
         .attr("r", 4)
```
Comment (jebking) [inline js/timelinevis.js]: magic number

**Review by jebking — CHANGES_REQUESTED** [review]
awesome, but some stuff to note

**Comment by aplett98** [comment]
commit 67bd9f3 makes requested changes.

**Review by jebking — APPROVED** [review]
nice :)

**Pull request merged**